### PR TITLE
Search Widget Added

### PR DESCRIPTION
**Pull request by tibrahulz**
## Describe your changes
- Added Search Widget as per the requirement.

## Issue ticket number and link
https://github.com/EnsembleUI/ensemble-react/issues/23

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have added tests

### CODE REVIEW

**Comment by evshi** [comment]
@tibrahul please rebase this onto main to uptake the latest changes

File: apps/starter/src/ensemble/dashboard.yaml [hunk 1]
```diff
@@ -58,6 +58,10 @@ View:
                   crossAxis: stretch
                   margin: 0 20px
                   children:
+                    - Search:
+                        placeholder: Search
+                        data: data.getData.body.users
```
Comment (evshi) [inline apps/starter/src/ensemble/dashboard.yaml:63]: You no longer need the `data` prefix, you can use `getData` directly

File: apps/starter/src/ensemble/dashboard.yaml [hunk 2]
```diff
@@ -58,6 +58,10 @@ View:
...
+                        searchKey: email
```
Comment (evshi) [inline apps/starter/src/ensemble/dashboard.yaml:64]: We will also need the ability to hook this up to API filtering and search params.

**Review by evshi — COMMENTED** [review]
Looks like you have some build issues to address also

File: packages/runtime/src/widgets/Widgets.css [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.no-box-shadow:focus {
+  box-shadow: none !important;
+}
```
Comment (evshi) [inline packages/runtime/src/widgets/Widgets.css:3]: Can we stick to using CSS-in-JS for now? We need to set up bundling stylesheets separately

File: packages/runtime/src/widgets/Search.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      );
+    }
+  };
+
+  return (
+    <AutoComplete
+      popupMatchSelectWidth={styles?.width}
+      options={options}
+      onSelect={() => {}}
+      onSearch={handleSearch}
```
Comment (evshi) [inline packages/runtime/src/widgets/Search.tsx:61]: Per our discussion, can you make a separate ticket to add event handling for Search?

**Review by evshi — COMMENTED** [review]
Thanks for taking this over! Let's just clean up the stylesheets for now and get this merged in

File: packages/runtime/src/widgets/Search.tsx (hunk 4 above)
Comment (anserwaseem) [reply packages/runtime/src/widgets/Search.tsx:60]: Its here https://github.com/EnsembleUI/ensemble-react/issues/96

**Review by evshi — APPROVED** [review]

**Pull request merged**